### PR TITLE
[feature-wip](Cloud) Support Azure Cloud Blob in FE

### PR DESCRIPTION
**Pull request by ByteYue**
## Proposed changes

As one subsequent pr of #35307, this pr tries to support Azure Cloud in FE. After this pr, we should be able to use Azure Blob for Cold Heat Separation and Storage Compute Separation along with backup restore to Azure Cloud.

For the data lakes support, you can add corresponding sdks to FE and implements the corresponding features.

### CODE REVIEW

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

File: regression-test/suites/tpcds_sf100_dup_without_key_p2/load.groovy [hunk 1]
```diff
@@ -32,7 +32,8 @@ suite('load') {
         |"AWS_ACCESS_KEY" = "${getS3AK()}",
         |"AWS_SECRET_KEY" = "${getS3SK()}",
         |"AWS_ENDPOINT" = "${getS3Endpoint()}",
-        |"AWS_REGION" = "${getS3Region()}")
+        |"AWS_REGION" = "${getS3Region()}"),
```
Comment (gavinchou) [inline regression-test/suites/tpcds_sf100_dup_without_key_p2/load.groovy:35]: bug , should not end if `),`

**Comment by gavinchou** [comment]
Add a case to  cover s3 load without `provider` option
```
oad ....
WITH S3 (
        "AWS_ACCESS_KEY" = "aaa",
        "AWS_SECRET_KEY" = "bbb",
        "AWS_ENDPOINT" = "ccc",
        "AWS_REGION" = "ddd")
        PROPERTIES(
        "exec_mem_limit" = "8589934592",
        "load_parallelism" = "3")
```

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/fs/remote/AzureFileSystem.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    private void initFsProperties() {
+        this.properties.putAll(((AzureObjStorage) objStorage).getProperties());
+    }
+
+    @Override
+    protected FileSystem nativeFileSystem(String remotePath) throws UserException {
+        return null;
+    }
+
+    private static String removeUselessSchema(String remotePath) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/fs/remote/AzureFileSystem.java:66]: Add comment with an example to describe the transform

File: fe/fe-core/src/main/java/org/apache/doris/fs/remote/AzureFileSystem.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                                blobItem.getProperties().getContentLength(),
+                                blobItem.getProperties().getLastModified().getSecond());
+                        result.add(remoteFile);
+                    }
+                }
+                newContinuationToken = pagedResponse.getContinuationToken();
+            } while (newContinuationToken != null);
+
+        } catch (BlobStorageException e) {
+            LOG.info("file not found: " + e.getMessage());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/fs/remote/AzureFileSystem.java:118]: better print file path in log

File: fe/fe-core/src/main/java/org/apache/doris/fs/remote/AzureFileSystem.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            return new Status(Status.ErrCode.NOT_FOUND, "file not found: " + e.getMessage());
+        } catch (Exception e) {
+            LOG.error("errors while get file status ", e);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/fs/remote/AzureFileSystem.java:121]: better print file path in log.
And use warn

File: fe/fe-core/src/main/java/org/apache/doris/fs/obj/AzureObjStorage.java [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+    public Triple<String, String, String> getStsToken() throws DdlException {
+        return null;
+    }
+
+    @Override
+    public Status headObject(String remotePath) {
+        BlobClient blobClient = client.getBlobClient(remotePath);
+        try {
+            BlobProperties properties = blobClient.getProperties();
+            LOG.info("head file " + remotePath + " success: " + properties.toString());
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/fs/obj/AzureObjStorage.java:102]:
```suggestion
            LOG.info("head file {} success: {}", remotePath, properties.toString());
```

File: fe/fe-core/src/main/java/org/apache/doris/fs/obj/AzureObjStorage.java [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+            return Status.OK;
+        } catch (BlobStorageException e) {
+            if (e.getStatusCode() == HttpStatus.SC_NOT_FOUND) {
+                return new Status(Status.ErrCode.NOT_FOUND, "remote path does not exist: " + remotePath);
+            } else {
+                LOG.warn("headObject failed:", e);
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/fs/obj/AzureObjStorage.java:108]: print file path

File: fe/fe-core/src/main/java/org/apache/doris/catalog/AzureResource.java [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+            String bucketName = properties.get(S3Properties.BUCKET);
+            String rootPath = properties.get(S3Properties.ROOT_PATH);
+            pingS3(credential, bucketName, rootPath, properties);
+        }
+        // optional
+        S3Properties.optionalS3Property(properties);
+        this.properties = properties;
+    }
+
+    private static void pingS3(CloudCredentialWithEndpoint credential, String bucketName, String rootPath,
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/AzureResource.java:89]:
```suggestion
    private static void pingAzure(CloudCredentialWithEndpoint credential, String bucketName, String rootPath,
```

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Comment by ByteYue** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by morningman — APPROVED** [review]

**Pull request merged**